test/functional/feature_startupnotify.py

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test -startupnotify."""
+
+import os
+
+from test_framework.test_framework import BitcoinTestFramework
+
+
+class StartupNotifyTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.disable_syscall_sandbox = True
+        self.extra_args = [["-startupnotify=echo 'test' >> node0/test.txt"]]
+
+    def run_test(self):
+        tmpdir = os.path.join(self.options.tmpdir, "node0", "test.txt")
+
+        self.log.info("When node starts, check if test.txt exists")
```

Comment:
```suggestion
        self.log.info("Test -startupnotify command is run when node starts")
```

Reply:
Thank you for the suggestions, they are very relevant, I am going to implement them.